### DISCUSSION TITLE
Show span name instead of SQL statement?

### DISCUSSION THREAD

**Discussion by HeCorr** [uptrace/uptrace#324] [Q&A]
Hello! I'm playing around with Uptrace and noticed that database spans show the SQL statement as the name instead of the actual span name:

![image](https://github.com/uptrace/uptrace/assets/75134774/7905f153-4df6-4084-8c21-2c4928efe3ce)

If I change the `group by` filter to `_name` they show up incorrectly:

![image](https://github.com/uptrace/uptrace/assets/75134774/dc7f540d-a653-4b4e-8cc2-9b6ad9ee62e3)

Is there a way to change this?

**Comment by vmihailenco** [comment]
>If I change the group by filter to _name they show up incorrectly:

This is a bug. Thanks for reporting.

>show the SQL statement as the name instead of the actual span name

Usually, span name is rather useless (not in your case though.) so that's why we display the SQL.

Atm I don't have good ideas how to improve this, but we'll probably come up with something eventually...

**Comment by HeCorr** [comment]
> This is a bug. Thanks for reporting.

oh, that's interesting. you're welcome!

> Atm I don't have good ideas how to improve this

What about a simple per-project setting that makes the span name be shown instead? If you can tell me whether that logic happens in the back or frontend I'm willing to try something (I'm going to try it anyways but it might save me a bit of digging haha)

**Reply by HeCorr** [reply]
I think I found the [relevant code](https://github.com/uptrace/uptrace/blob/85cbc63106c23cceebc8f78a04d6e69be3df3d38/pkg/tracing/span_attrs.go#L395).

@vmihailenco do you think it's best for this setting to be per instance or per project? I think per project gives the best flexibility and is easier to implement since [`assignSpanSystemAndGroupID`](https://github.com/uptrace/uptrace/blob/85cbc63106c23cceebc8f78a04d6e69be3df3d38/pkg/tracing/span_attrs.go#L351) already has the project config as argument.

**Reply by vmihailenco** [reply]
@HeCorr Let's do it per project
